packages/ui/src/customRenderers/fileUpload/fileUpload.tsx

```diff
@@ -53,7 +58,13 @@ const FileUpload = forwardRef(function FileUpload(props: FileUploadProps, ref) {
 				};
 			}) as FormattedFile[];
 			props.onUpdateFiles(formattedFiles);
-		});
+		};
+		uppy.on("complete", handler);
+
+		// Make sure we only have one listener at a time
+		return () => {
+			uppy.off("complete", handler);
+		};
```

Comment:
without this, subscription seemed to happen twice since the component rerendered in the app, so the 'complete' handler would fire twice, resulting in double the images being inserted in the editor!